### PR TITLE
modem: pdn: Allow PDP_TYPE-only config

### PR DESCRIPTION
**Pull request by junqingzou**
Allow PDN configuration without APN but PDP_Type only.

Signed-off-by: Jun Qing Zou <jun.qing.zou@nordicsemi.no>

### CODE REVIEW

**Review by lemrey — APPROVED** [review]

**Comment by junqingzou** [comment]
@rlubos Can this be merged first?

**Review by lemrey — COMMENTED** [review]
@junqingzou do we need a changelog entry for this?

**Comment by junqingzou** [comment]
> @junqingzou do we need a changelog entry for this?

I'll rebase and add changelog.
Fortuantely PDN_Type is not the failure cause for ZZHC but still good to have this feature.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -301,7 +301,9 @@ Debug libraries
 Modem libraries
 ---------------
 
-|no_changes_yet_note|
+* :ref:`pdn_readme` library:
+
+  * Updated the library to allow PDP_TYPE only configuration in the :c:func:`pdn_ctx_configure` function.
```
Comment (wiba-nordic) [inline doc/nrf/releases/release-notes-changelog.rst:306]: What is PDP_TYPE? It seems like a variable, but a quick search doesn't turn up anything?

**Review by lemrey — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (junqingzou) [reply doc/nrf/releases/release-notes-changelog.rst:306]: Please refer to [link](https://infocenter.nordicsemi.com/topic/ref_at_commands/REF/at_commands/packet_domain/cgdcont_set.html?cp=2_1_6_0_0)

Comment (wiba-nordic) [inline doc/nrf/releases/release-notes-changelog.rst:306]:
```suggestion
  * Updated the library to allow a ``PDP_TYPE``-only configuration in the :c:func:`pdn_ctx_configure` function.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (junqingzou) [reply doc/nrf/releases/release-notes-changelog.rst:306]: OK, I'll use "PDP_type" to align with AT manual in exact way

**Pull request merged**